src/common/cluster-store.ts

```diff
@@ -357,12 +357,11 @@ export const clusterStore = ClusterStore.getInstance<ClusterStore>();
 export function getClusterIdFromHost(host: string): ClusterId | undefined {
   // e.g host == "%clusterId.localhost:45345"
   const subDomains = host.split(":")[0].split(".");
+  const lastTwoParts = subDomains.slice(-2);
 
-  if (subDomains.length >= 2) {
-    return subDomains[0];
-  }
+  lastTwoParts.pop(); // remove last entry (namely "localhost")
 
-  return undefined;
+  return lastTwoParts[0]; // ClusterId or undefined
```

Comment:
```suggestion
  return subDomains.slice(-2, -1); // ClusterId or undefined
```

Reply:
I guess I could do `subDomains.slice(-2, -1)[0]` if you really want.